### ISSUE TITLE
npm plugin: issues with legacyAuth

### ISSUE THREAD

**Issue by sethomas** [intuit/auto#1734]
**Describe the bug**

Setting `legacyAuth` to true in `auto` config, does not properly send the value of `_auth` to `npm publish`.

**To Reproduce**

1. setup a npm project to use `auto` and the `npm` plugin
1. set `legacyAuth` to `true` in the plugin field
1. set `publishConfig.registry` in your `package.json` to be an external registry
1. make a `.env` file, add in `NPM_TOKEN`
1. run `npx auto canary` and see below error:
```
Error: Running command 'npm' with args [publish, --tag, canary, --_auth, MASKED_NPM_TOKEN] failed

npm ERR! code ERR_INVALID_ARG_TYPE
npm ERR! The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type boolean
```

(`MASKED_NPM_TOKEN` being the actual value from `.env`'s `NPM_TOKEN`)

**Expected behavior**

Should be passing the token correctly to `npm publish`.

**Additional context**

Looking at [this](https://github.com/intuit/auto/blob/master/plugins/npm/src/index.ts#L215-L218) block of code, it passes to `execPromise`, and the token is considered an argument and not the value.

If I changed those lines to the below, all is functional:
```
return [
    options.isMonorepo ? "--legacy-auth" : "--_auth" + `=${process.env.NPM_TOKEN}`,
];
```

**Comment by hipstersmoothie** [comment]
Would you be willing to submit a PR? @sethomas

**Comment by sethomas** [comment]
@hipstersmoothie yeah absolutely, got it all ready in my local but I can't push up

**Comment by hipstersmoothie** [comment]
Why can't you push?

**Comment by sethomas** [comment]
`git push origin npm-legacy-auth`:
```
remote: Permission to intuit/auto.git denied to sethomas.
fatal: unable to access 'https://github.com/intuit/auto/': The requested URL returned error: 403
```

**Comment by sethomas** [comment]
ah you want me to fork, one moment

**Comment by adierkens** [comment]
:rocket: Issue was released in [`v10.11.0`](https://github.com/intuit/auto/releases/tag/v10.11.0) :rocket:

**Comment by adierkens** [comment]
:rocket: Issue was released in [`v10.11.0`](https://github.com/intuit/auto/releases/tag/v10.11.0) :rocket:
